### ISSUE TITLE
UX tweaks to asset selector component in the entity panel 

### ISSUE THREAD

**Issue by jiajasper** [playcanvas/editor#711]
![image](https://user-images.githubusercontent.com/30747200/158400296-c6e82d68-f00d-40de-aef9-f77da4410c57.png)

This Character asset selector component could need some UX work.

### **Issues with this UX:**
1. Pen icon can be confused with a name change interaction
2. When there is no available asset to be selected, the asset panel has no clear visual cue to teach the user that they do not have the files of the correct type in their asset folders. 
3. This UX made me think that there was some sort of interactivity built between the asset panel and the entity panel. I attempted to drag files into the selector component and realised they were not linked (unlike lens studio or unity where you could drag target entities into the selector components)

### **Suggestions:**

**Quick**: Add an empty state in the asset panel - could be just a quick copy like "No available files found"
<img width="1029" alt="image" src="https://user-images.githubusercontent.com/30747200/158404438-2d9b3ffc-9a7d-4494-b637-47fc9327aec3.png">

**Medium**: Change the component a dropdown component where we filtered the right files for the users to select from - we are already doing the filtering with the current approach, but we can use a different UI to feed the options to the users.
<img width="668" alt="image" src="https://user-images.githubusercontent.com/30747200/158404035-97a75462-54b1-4d33-b54d-a8f3178be17c.png">

**Heavy**: Enable dragging and dropping assets into selector components in the entity panel (or even straight into the scene).

**Comment by jiajasper** [comment]
@yaustar yes, but this component is actually used elsewhere as well, for example 

<img width="330" alt="image" src="https://user-images.githubusercontent.com/30747200/158405985-e8540288-3559-4f91-a6c4-c4a3ff9aba36.png">

**Comment by yaustar** [comment]
Yeah, the reason I asked is that drag and drop is already implemented :) But we have an issue that clicking on anything else in the assets panel while the anim state graph is up, closes the state graph 😅 

https://user-images.githubusercontent.com/16639049/158407483-d97c2391-8d74-4d1a-8923-492191af2c70.mp4


> ![image](https://user-images.githubusercontent.com/16639049/158406977-189601f4-b3f8-49be-b335-3359bb5da2f9.png)

Might be best to show the full paths for the assets as a user can have multiple assets with the same name but in different folders. It is also possible to have two assets in the same folder with the same name too because we still/reference assets by a unique id. 

A side issue with having the dropdown is that users cannot navigate to the asset via the panel unless we allow both options

https://user-images.githubusercontent.com/16639049/158407940-dedc5ad3-c1f5-4208-acf8-d65ce896deae.mp4

**Comment by jiajasper** [comment]
@yaustar ahh i see, it makes sense. I think a quick win is by adding a quick line of copy in the asset panel when no assets could be used - since this is the point of user journey that could cause some confusion. For the rest, i think we are good for now!

**Comment by yaustar** [comment]
For this ticket, we should only do the quick version where we show an empty state if there are no available assets in the current filter/folder.

Perhaps we should change the message to ask them to look in another folder or change the search.
